libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/TokenUtil.java

```diff
@@ -30,8 +33,19 @@ public class TokenUtil {
    * @param publicKey the public key
    * @param token     the token
    */
-  public static void convertToken(UserContext userContext, String publicKey, String token) {
-    Map<String, Object> tokenValuesMap = ContextUtils.getTokenValues(publicKey, token);
+  public static void convertToken(UserContext userContext, String publicKey, JwtClassicConfig jwtClassicConfig, String token) {
+
+    Map<String, Object> tokenValuesMap = null;
+    if(publicKey != null) {
+      tokenValuesMap = ContextUtils.getTokenValues(publicKey, token);
+    }
+    if(jwtClassicConfig != null) {
+      try {
+        tokenValuesMap = ContextUtils.getTokenValues(jwtClassicConfig, token);
```

Comment:
Is there a possible a case where "publicKey" and "jwtClassicConfig" are both distinct from null?
 In that case, we are definining first tokenValuesMap using the publicKey, and then we are defining it again using the jwtClassicConfig. Therefore, we are assigning two times the value of the field. To avoid this, we can use an "if-else".

 If publickKey and jwtClassicConfig can not be both distinct from null at the same time, then ignore this comment 😬

Reply:
I made a change to fix your suggested change